gcloud/pubsub/_gax.py

```diff
@@ -180,7 +181,279 @@ def topic_list_subscriptions(self, topic_path):
+        except GaxError as exc:
+            if exc_to_code(exc.cause) == StatusCode.NOT_FOUND:
+                raise NotFound(subscription_path)
+            raise
+        return [_received_message_pb_to_mapping(rmpb)
+                for rmpb in response_pb.received_messages]
+
+    def subscription_acknowledge(self, subscription_path, ack_ids):
+        """API call:  acknowledge retrieved messages
+
+        See:
+        https://cloud.google.com/pubsub/reference/rest/v1/projects.subscriptions/modifyPushConfig
+
+        :type subscription_path: string
+        :param subscription_path: the fully-qualified path of the new
+                                  subscription, in format
+                                  ``projects/<PROJECT>/subscriptions/<SUB_NAME>``.
+
+        :type ack_ids: list of string
+        :param ack_ids: ack IDs of messages being acknowledged
+        """
+        try:
+            self._gax_api.acknowledge(subscription_path, ack_ids)
+        except GaxError as exc:
+            if exc_to_code(exc.cause) == StatusCode.NOT_FOUND:
+                raise NotFound(subscription_path)
+            raise
+
+    def subscription_modify_ack_deadline(self, subscription_path, ack_ids,
+                                         ack_deadline):
+        """API call:  update ack deadline for retrieved messages
+
+        See:
+        https://cloud.google.com/pubsub/reference/rest/v1/projects.subscriptions/modifyAckDeadline
+
+        :type subscription_path: string
+        :param subscription_path: the fully-qualified path of the new
+                                  subscription, in format
+                                  ``projects/<PROJECT>/subscriptions/<SUB_NAME>``.
+
+        :type ack_ids: list of string
+        :param ack_ids: ack IDs of messages being acknowledged
+
+        :type ack_deadline: int
+        :param ack_deadline: the deadline (in seconds) by which messages pulled
+                            from the back-end must be acknowledged.
+        """
+        try:
+            self._gax_api.modify_ack_deadline(
+                subscription_path, ack_ids, ack_deadline)
+        except GaxError as exc:
+            if exc_to_code(exc.cause) == StatusCode.NOT_FOUND:
+                raise NotFound(subscription_path)
+            raise
+
+
 def _message_pb_from_dict(message):
     """Helper for :meth:`_PublisherAPI.topic_publish`."""
     return PubsubMessage(data=_to_bytes(message['data']),
                          attributes=message['attributes'])
+
+
+def _subscription_pb_to_mapping(sub_pb):
+    """Helper for :meth:`list_subscriptions`, et aliae
+
+    Ideally, would use a function from :mod:`protobuf.json_format`, but
+    the right one isn't public.  See:
+    https://github.com/google/protobuf/issues/1351
+    """
+    mapping = {
+        'name': sub_pb.name,
+        'topic': sub_pb.topic,
+        'ack_deadline': sub_pb.ack_deadline,
+    }
+    if sub_pb.push_config.push_endpoint != '':
```

Comment:
https://cloud.google.com/pubsub/reference/rpc/google.pubsub.v1#google.pubsub.v1.PushConfig defines the type as `string`.  Should we be comparing to `u''`?